TCAT/Utilities/Network.swift

```diff
@@ -155,49 +159,46 @@ class AllBusStops: JSONDecodable {
     }
 }
 
-class AllBusLocations: JSONDecodable {
+class BusLocationResult: JSONDecodable {
 
-    var allBusLocations : [BusLocation] = [BusLocation]()
+    var busLocation: BusLocation? = nil
 
     required init(json: JSON) throws {
-        
         if json["success"].boolValue {
-            let data = json["data"].arrayValue
-            allBusLocations = parseAllLocations(json: data)
+            let data = json["data"]
+            busLocation = parseBusLocation(json: data)
         }
     }
 
-    func parseAllLocations(json: [JSON]) -> [BusLocation] {
-
-        var allLocationsArray = [BusLocation]()
-
-        for bus in json {
-
-            let routeID = bus["routeID"].stringValue
-            let busLocation = BusLocation(routeID: routeID)
-
-            busLocation.destination = bus["destination"].stringValue
-            busLocation.deviation = bus["deviation"].intValue
-            busLocation.direction = bus["direction"].stringValue
-            busLocation.displayStatus = bus["displayStatus"].stringValue
-            busLocation.gpsStatus = bus["gpsStatus"].intValue
-            busLocation.heading = bus["heading"].intValue
-            busLocation.lastStop = bus["lastStop"].stringValue
-            busLocation.lastUpdated = Date(timeIntervalSince1970: bus["lastUpdated"].doubleValue)
-            busLocation.latitude = bus["latitude"].doubleValue
-            busLocation.longitude = bus["longitude"].doubleValue
-            busLocation.name = bus["name"].intValue
-            busLocation.opStatus = bus["opStatus"].stringValue
-            busLocation.runID = bus["runID"].intValue
-            busLocation.speed = bus["speed"].intValue
-            busLocation.tripID = bus["tripID"].intValue
-            busLocation.vehicleID = bus["vehicleID"].intValue
-
-            allLocationsArray.append(busLocation)
-
-        }
-
-        return allLocationsArray
+    func parseBusLocation(json: JSON) -> BusLocation {
+        
+        let routeID = json["routeID"].stringValue
+        let busLocation = BusLocation(routeID: routeID)
+        busLocation.dataType = {
+            switch json["case"] {
```

Comment:
shouldn't there be a `.stringValue`?

Reply:
What do you mean?

Reply:
shouldn't it be `json["case"].stringValue`?